src/ExifViewer.jl

```diff
@@ -1,6 +1,153 @@
 module ExifViewer
 
 include("../lib/LibExif.jl")
+include("tags.jl") # for simple names
+
 using .LibExif
 
+export read_metadata, read_tags
+
+## Tasks 
+# We want to read from IO, data, filepath : done
+# We want to give option of reading specific id and specific tags too
+# We want to make it fast
+# We want docs and we want tests
+# We want to ensure pointer gets removed after information
+function numifds(ed::LibExif._ExifData)
+    return length(ed.ifd)
+end
+
+function numentriesinifd(data::LibExif._ExifContent)
+    return Int(data.count)
+end
+
+
+
+"""
+    read_metadata(data::Vector{UInt8}; kwargs...)
+    read_metadata(filepath::AbstractString; kwargs...)
+    read_metadata(io::IO; kwargs...)
+
+Reads EXIF data from a filepath, IO, vector{Int8} and returns a Dict with tags and their values.
+
+# Examples
+```jldoctest
+julia> using BenchmarkTools, TestImages, ExifViewer
+julia> filename = "test/test_images/test.jpg"
+
+julia> io = open(filename, "r")
+julia> read_metadata(io)
+Dict{Any, Any} with 12 entries:
+  "EXIF_TAG_DOCUMENT_NAME"     => "Test_Image"
+  "EXIF_TAG_ORIENTATION"       => "Bottom-right"
+  ⋮                            => ⋮
+
+julia> read_metadata(filename)
+Dict{Any, Any} with 12 entries:
+  "EXIF_TAG_DOCUMENT_NAME"     => "Test_Image"
+  "EXIF_TAG_ORIENTATION"       => "Bottom-right"
+  ⋮                            => ⋮
+
+julia> a = read("test/test_images/test.jpg")
+julia> read_metadata(a)
+Dict{Any, Any} with 12 entries:
+  "EXIF_TAG_DOCUMENT_NAME"     => "Test_Image"
+  "EXIF_TAG_ORIENTATION"       => "Bottom-right"
+  ⋮                            => ⋮
+
+```
+
+
+```julia
+julia> using BenchmarkTools, TestImages, ExifViewer
+julia> filename = "test/test_images/test.jpg"
+
+julia> io = open(filename, "r")
+julia> @btime read_metadata(io)
+1.862 μs (11 allocations: 912 bytes)
+
+julia> @btime read_metadata(filename)
+269.472 μs (122 allocations: 100.05 KiB)
+
+julia> @btime data = read(filename)
+15.876 μs (13 allocations: 69.28 KiB)
+
+julia> @btime read_metadata(data)
+221.786 μs (109 allocations: 30.77 KiB)
+```
```

Comment:
There's no need to show the performance benchmark in the docstring. If the performance is an issue, open an github issue for it and leave a comment in the source code should be sufficient.